### PR TITLE
[DRAFT] Merge 3.3.6

### PR DESCRIPTION
**Pull request by kball**
## What does this PR do?

Brings us up to date with Cal's release version 3.3.6.  (They just released 3.3.7 but it breaks typescript checks just on a vanilla pull so not merging it in just yet)

## Test Plan

Tests from notion:

- [ ] Onboarding - set user in db to onboarding false an ensure going through onboarding syncs user-name profile stuff to Application DB
- [x] Schedule Single Session - make sure single session is found in Cal, Google Cal and also in DB sessions
- [ ] Cancel Single Session -  make sure single session is cancelled in Cal, removed from Google Cal and also cancelled DB sessions
- [x] Reschedule Single Session - make sure single session is reschedule in Cal, moved/updated in Google Cal and also updated in DB sessions
- [x] Schedule Recurring Session - make sure multiple sessions is found in Cal, Google Cal and also in DB sessions
- [x] Cancel Single Recurring Session -  make sure single session is cancelled in Cal, removed from Google Cal and also cancelled DB sessions
- [x] Cancel Remaining Recurring Session -  make sure remaining session are cancelled in Cal, removed from Google Cal and also cancelled DB sessions
- [ ] Reschedule Single Session - make sure single session is reschedule in Cal, moved/updated in Google Cal and also updated in DB sessions

Have not done these yet - will update from draft status after I do

### CODE REVIEW

**Comment by kball** [comment]
Rescheduling recurring is kind of broken/weird in this release, but I want to get Jacob's eyes on it, and other parts seem to all be working so I'm inclined to get this into staging. 

@dgalindez what do you think?

**Comment by dgalindez** [comment]
> Rescheduling recurring is kind of broken/weird in this release, but I want to get Jacob's eyes on it, and other parts seem to all be working so I'm inclined to get this into staging.
> 
> @dgalindez what do you think?

I'd push to staging. From experience a lot of the bugs don't show up until we have more heavy usage than local

File: .env.example [hunk 1]
```diff
@@ -59,6 +59,11 @@ SAML_CLIENT_SECRET_VERIFIER=
 # PGSSLMODE='no-verify'
 PGSSLMODE=
 
+# Define which hostnames are expected for the app to work on
+ALLOWED_HOSTNAMES='"cal.com","cal.dev","cal-staging.com","cal.community","cal.local:3000","localhost:3000"'
+#  Reserved orgs subdomains for our own usage
+RESERVED_SUBDOMAINS='"app","auth","docs","design","console","go","status","api","saml","www","matrix","developer","cal","my","team","support","security","blog","learn","admin"'
```
Comment (dgalindez) [inline .env.example:65]: Just to confirm, both of this have been added to the deployment variables and have adequate values for us? e.g. `app.mento.co` is the app, and it seems to be referenced here
Also on `ALLOWED_HOSTNAMES` we'd need `cal.mento.co` correct?

File: apps/ai/src/tools/getAvailability.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const params: { [k: string]: string } = {
+    apiKey,
+    userId: userId.toString(),
+    dateFrom,
+    dateTo,
+  };
+
+  const urlParams = new URLSearchParams(params);
+
+  const url = `${env.BACKEND_URL}/availability?${urlParams.toString()}`;
```
Comment (dgalindez) [inline apps/ai/src/tools/getAvailability.ts:30]: Seems like we should be able to hit this BE if we ever move to only wanting availability from cal

**Review by dgalindez — COMMENTED** [review]
Leaving some comments but won't review all just yet, it is a bunch of code

**Pull request merged**